components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/util/TemplateManagerHelper.java

```diff
@@ -235,35 +193,51 @@ public static void validateTemplateGroup(TemplateGroup templateGroup) throws Tem
      * - Validate all properties
      *
-     * @param ruleTemplate
-     * @throws TemplateManagerException
+     * @param ruleTemplate rule template object
+     * @throws TemplateManagerException template manager exceptions
      */
-    public static void validateRuleTemplate(RuleTemplate ruleTemplate) throws TemplateManagerException {
-        try {
+    private static void validateRuleTemplate(RuleTemplate ruleTemplate) throws TemplateManagerException {
             if (ruleTemplate.getName() == null) {
                 throw new TemplateManagerException("Invalid rule template - Rule template name is null ");
             }
+            if (ruleTemplate.getName().isEmpty()) {
+                throw new TemplateManagerException("Invalid rule template - Rule template name is empty ");
+            }
             if (ruleTemplate.getUuid() == null) {
                 throw new TemplateManagerException("Invalid rule template - UUID is null for rule template : " +
                         ruleTemplate.getName());
             }
-            if (ruleTemplate.getInstanceCount().equals(TemplateManagerConstants.INSTANCE_COUNT_ONE) ||
-                    ruleTemplate.getInstanceCount().equals(TemplateManagerConstants.INSTANCE_COUNT_MANY)) {
-                if (ruleTemplate.getType() == null) {
-                    throw new TemplateManagerException("Invalid rule template - " +
-                            "rule template type is null for rule template : " +
-                            ruleTemplate.getUuid());
-                }
+            if (ruleTemplate.getUuid().isEmpty()) {
+                throw new TemplateManagerException("Invalid rule template - UUID is empty for rule template : " +
+                        ruleTemplate.getName());
             }
-            if (!(ruleTemplate.getType().equals(TemplateManagerConstants.RULE_TEMPLATE_TYPE_TEMPLATE) ||
-                    ruleTemplate.getType().equals(TemplateManagerConstants.RULE_TEMPLATE_TYPE_INPUT) ||
-                    ruleTemplate.getType().equals(TemplateManagerConstants.RULE_TEMPLATE_TYPE_OUTPUT))) {
+            if (ruleTemplate.getInstanceCount()==null){
+                throw new TemplateManagerException("Invalid rule template - Instance count field is null in " +
+                        "ruleTemplate : " + ruleTemplate.getName());
+            }
+            if (!(ruleTemplate.getInstanceCount().toLowerCase().equals(TemplateManagerConstants.INSTANCE_COUNT_ONE) ||
+                    ruleTemplate.getInstanceCount().toLowerCase().equals(TemplateManagerConstants
+                            .INSTANCE_COUNT_MANY))) {
+                throw  new TemplateManagerException("Invalid rule template - Instance count field should be either " +
+                        "'one' or 'many' in ruleTemplate : " + ruleTemplate.getName());
+            }
+            if (ruleTemplate.getType() ==null){
```

Comment:
format the code properly in all palces

Reply:
Fixed by 52ac39524ce454695682a772fd3964637a8dffde